polynote-frontend/polynote/util/helpers.ts

```diff
@@ -152,6 +180,41 @@ export function collect<T, U>(arr: T[], fun: (t: T) => U | undefined | null): U[
     })
 }
 
+export function collectMatch<T, R>(arr: T[], fn: (matcher: Matcher<T>) => Matcher<T, R>): R[] {
```

Comment:
This is very minor, but do you think there's any benefit for these `collectX` functions to be implemented using `collect`? Did that not work for some reason?